lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1373,61 +1363,10 @@ newtype Signature (what :: Type) = Signature { getSignature :: ByteString }
     deriving stock (Show, Eq, Generic)
     deriving newtype (ByteArrayAccess)
 
--- | A polymorphic wrapper type with a custom show instance to display data
--- through 'Buildable' instances.
-newtype ShowFmt a = ShowFmt { unShowFmt :: a }
-    deriving (Generic, Eq, Ord)
-
-instance NFData a => NFData (ShowFmt a)
-
-instance Buildable a => Show (ShowFmt a) where
-    show (ShowFmt a) = fmt (build a)
-
--- | Checks whether or not an invariant holds, by applying the given predicate
---   to the given value.
---
--- If the invariant does not hold (indicated by the predicate function
--- returning 'False'), throws an error with the specified message.
---
--- >>> invariant "not empty" [1,2,3] (not . null)
--- [1, 2, 3]
---
--- >>> invariant "not empty" [] (not . null)
--- *** Exception: not empty
-invariant
-    :: String
-        -- ^ The message
-    -> a
-        -- ^ The value to test
-    -> (a -> Bool)
-        -- ^ The predicate
-    -> a
-invariant msg a predicate =
-    if predicate a then a else error msg
-
```

Comment:
Nice idea to move these utility functions to a dedicated module.

(The `Types` module is already bloated enough, so moving these out is good IMO.)